### DISCUSSION TITLE
Request: Allow contribution of options for GlspVscodeConnector - Messenger

### DISCUSSION THREAD

**Discussion by jmenzies12** [eclipse-glsp/glsp#1454] [General] [answered]
Hey there,

The VSCode Messenger has an option called 'ignoreHiddenViews', this is set to true when constructed without passing in any options. This stops the diagram client from receiving any action messages etc. whenever the webview is considered hidden, for example if you have it as one of your tabs but it is not the editor in focus.

This causes issues in our project whenever someone changes tab to edit our source model, as the message sent from the source model to the webview never arrives at the client, resulting in the model not updating.

Unfortunately Messenger is declared as read-only in GlspVscodeConnector, so I had to copy GlspVscodeConnector and GlspEditorProvider in order to pass the options in for this.

It would be great if you could add a way for us to provide this in future.

Kind Regards

**Comment by tortmayr** [comment, marked as answer]
@jmenzies12 
That's a very reasonable request. I will convert this discussion into an issue so that we can tackle it in the next release cycle.

**Reply by jmenzies12** [reply]
Great, thank you!

**Reply by tortmayr** [reply]
I finally managed to take a look at this. I will provide a PR soon to make this configurable and also use  `ignoreHiddenViews`=false as default.

Sidenote: Instead of copying the `GLSPVscodeConnector` you could also make use of the `Writable` utility type provided by GLSP that allows you to bypass the readonly restriction.

```ts
export const messenger = new Messenger({ ignoreHiddenViews: false });
const connector = new GlspVscodeConnector(...);
(connector as Writable<GlspVscodeConnector>).messenger = messenger;
 ```

**Reply by jmenzies12** [reply]
Hey @tortmayr,

Thanks for looking into this! We ended up setting ignoreHiddenViews to true, in our case it resulted in a bit more reloading than we wanted.

Instead we decided to keep track of reload requests in the webview and then added a listener to handle "onDidChangeViewState", when the user comes back to the diagram and there has been an external reload request, it will send a reloadModelAction to the GLSP Client.

I do think adding the ability to make ignoreHiddenViews configurable is great regardless.
